packages/bpk-component-autosuggest/src/BpkAutosuggest.tsx

```diff
@@ -0,0 +1,408 @@
+      onSuggestionSelected,
+      onSuggestionsClearRequested,
+      onSuggestionsFetchRequested,
+      renderBesideInput,
+      renderSectionTitle,
+      renderSuggestion,
+      shouldRenderSuggestions,
+      showClear,
+      suggestions,
+      theme: customTheme,
+    },
+    forwardedRef,
+  ) => {
+    const theme = { ...defaultTheme, ...customTheme };
+
+    function stateReducer(
+      state: UseComboboxState<any>,
+      actionAndChanges: UseComboboxStateChangeOptions<any>,
+    ) {
+      const { changes, type } = actionAndChanges;
+      switch (type) {
+        case useCombobox.stateChangeTypes.InputFocus:
+          return {
+            ...changes,
+            isOpen: false, // keep the menu closed when input gets focused.
+          };
+        default:
+          return changes;
+      }
+    }
+    const {
+      getInputProps,
+      getItemProps,
+      getLabelProps,
+      getMenuProps,
+      highlightedIndex,
+      inputValue,
+      isOpen,
+      openMenu,
+      setInputValue,
+    } = useCombobox({
+      stateReducer,
+      items: suggestions,
+      itemToString(suggestion: any | null) {
+        return suggestion ? getSuggestionValue(suggestion) : '';
+      },
+      async onInputValueChange(changes) {
+        onInputValueChange?.({
+          method: changes.type,
+          newValue: changes.inputValue ?? '',
+        });
+
+        if (changes.inputValue?.length) {
+          if (changes.isOpen) {
+            onSuggestionsFetchRequested(changes.inputValue);
+          }
+        } else if (suggestions.length) {
+          onSuggestionsClearRequested();
+        }
+      },
+      onSelectedItemChange(changes) {
+        if (changes.selectedItem) {
+          setInputValue(getSuggestionValue(changes.selectedItem));
+          onSuggestionSelected?.({
+            suggestion: changes.selectedItem,
+            inputValue,
+          });
+        }
+      },
+      getA11yStatusMessage({ resultCount }) {
+        return getA11yResultsMessage?.(resultCount) ?? '';
+      },
+      initialInputValue: defaultValue ?? '',
+      id,
+    });
+
+    useEffect(() => {
+      if (defaultValue) {
+        setInputValue(defaultValue);
+      }
+    }, [defaultValue, setInputValue]);
+
+    useEffect(() => {
+      if (isBanana && !isDesktop) {
+        onLoad?.(inputValue);
+      }
+      // fire track event on load and forget about it after. We don't want to track again when anything (inputValue) changes
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    const onClickOrKeydown = () => {
+      if (shouldRenderSuggestions) {
+        shouldRenderSuggestions(inputValue);
+        openMenu();
+      }
+
+      if (!isOpen && inputValue.length) {
+        onSuggestionsFetchRequested(inputValue);
+        openMenu();
+      } else {
+        onClick?.();
+      }
+
+      // Desktop destination autosuggest lives on the homepage and is "loaded/interacted with" via clicking on it
```

Comment:
Not sure I understand this, might be useful to have a chat separately, but I'm also thinking whether we could use something else other than `isBanana` for this boolean. Why is there a special behaviour for when the component is render in banana?

Reply:
Yea you're right the `isBanana` flag use was wrong here. I'll add a couple of tests to help with the understanding.